[WIP] Add lowest and highest dependencies tests and fix errors

tgalopin opened the pull request:
As proposed in puli/issues#110, it's a good idea to test the lowest and highest dependecies as well as the current ones.

tgalopin commented:
The version in composer.json needs to be adjusted.

tgalopin commented:
Tests are failing due to the missing TestUtil class (and perhaps other things but we can't debug them for the moment).

webmozart commented on composer.json:
```diff
@@ -11,15 +11,16 @@
     "require": {
         "php": ">=5.3.9",
-        "puli/repository": "1.0.*",
+        "puli/repository": "^1.0",
```
I think that this change will cause problems, since Composer will try to update to 1.1-dev (which is not actively maintained yet).

webmozart commented on .travis.yml:
```diff
@@ -1,25 +1,30 @@
+    - php: 5.4
+    - php: 5.5
+    - php: 5.6
+    - php: 5.6
```
One 5.6 too much

tgalopin replied:
Removed

webmozart commented:
@stof What do you think about this PR? Should we apply the same fix to all other repositories?

tgalopin commented:
I fixed the tests and I updated a bit the travis.yml file.

@stof do you think of changes you would make to the travis.yml file before we copy it to all the other repositories? Especially, what do you think of adding a build for deps=low in PHP5.3? Useless or useful?

tgalopin commented:
This is ready for merge in my opinion.
